irohad/validation/impl/stateful_validator_impl.cpp

```diff
@@ -53,41 +71,55 @@ namespace iroha {
       // Filter only valid transactions
       auto filter = [&temporaryWsv, checking_transaction](auto &acc,
                                                           const auto &tx) {
-        auto answer = temporaryWsv.apply(tx, checking_transaction);
+        auto *old_tx = tx->makeOldModel();
```

Comment:
better to use unique_ptr, always deallocated (even at the exception) and almost no overhead